agents-api/agents_api/activities/tool_executor.py

```diff
@@ -0,0 +1,231 @@
+import json
+from typing import Any, Literal
+
+from pydantic import BaseModel, Field
+
+from ..clients.integrations import run_integration_service
+from ..env import brave_api_key
+
+
+class ToolExecutionResult(BaseModel):
+    """Represents the result of executing a tool"""
+
+    id: str
+    name: str | None = None
+    output: dict[str, Any]
+    error: str | None = None
+
+
+class WebSearchToolCall(BaseModel):
```

Comment:
This conflicts with another model in `agents-api/agents_api/autogen/Responses.py`. Please give this another name.